client/src/main/java/org/apache/falcon/resource/EntitySummaryResult.java

```diff
@@ -67,6 +67,24 @@ public void setEntitySummaries(EntitySummary[] entitySummaries) {
         this.entitySummaries = entitySummaries;
     }
 
+    @Override
+    public Object[] getCollection() {
+        return entitySummaries;
+    }
+
+    @Override
+    public void setCollection(Object[] items) {
+        if (items == null) {
+            setEntitySummaries(new EntitySummary[0]);
+        } else {
+            EntitySummary[] summaries = new EntitySummary[items.length];
```

Comment:
Instead of iterating over array and casting each element, you can always do -

EntitySummary[] summaries = Arrays.copyOf(items, items.length, EntitySummary[].class);